internal/expr/prepare.go

```diff
@@ -1,43 +1,46 @@
 package expr
 
 import (
-	"bytes"
 	"fmt"
 	"reflect"
 	"sort"
-	"strconv"
 	"strings"
 )
 
-// PreparedExpr contains an SQL expression that is ready for execution.
+// PreparedExpr represents a valid SQLair statement.
 type PreparedExpr struct {
-	outputs []typeMember
-	inputs  []typeMember
-	sql     string
+	preparedParts []preparedPart
 }
 
-// SQL returns the SQL ready for execution.
-func (pe *PreparedExpr) SQL() string {
-	return pe.sql
+// preparedPart represents a part of a SQLair statement. It contains
+// information to generate the relevent SQL and access any Go types pertaining
+// to the part.
+type preparedPart interface {
+	preparedPart()
 }
 
-const markerPrefix = "_sqlair_"
+type preparedOutput struct {
+	queryColumns []columnAccessor
+	outputValues []typeMember
```

Comment:
[Nit] What do you think of naming it `outputTypes` or `columnTypes`.

Reply:
I went with `outputValues` because we have got `valueAccessor`. But you're right, they're not really values. I might just switch it to `outputs` since that's what it was before. I think the problem with calling them types is that they are not really types, they are struct fields, or map keys which is slightly different. I think maybe `outputLocations` could work as well if we wanted to be more specific.

Reply:
I think it would be good to discuses this further, I think we still haven't found a great consistent name for this concept across the codebase.